`ExtensionDescriptorMojoTest` fails on a pristine computer

### Describe the bug

The `ExtensionDescriptorMojoTest` fails on a pristine system with an error like this:

```
[ERROR] io.quarkus.maven.ExtensionDescriptorMojoTest.shouldProcessRealisticExtensionCleanly -- Time elapsed: 8.031 s <<< ERROR!
org.apache.maven.plugin.MojoExecutionException: Failed to resolve dependencies of io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT
    at io.quarkus.maven.ExtensionDescriptorMojo.resolveRuntimeDeps(ExtensionDescriptorMojo.java:890)
    at io.quarkus.maven.ExtensionDescriptorMojo.validateExtensionDeps(ExtensionDescriptorMojo.java:788)
    at io.quarkus.maven.ExtensionDescriptorMojo.execute(ExtensionDescriptorMojo.java:255)
    at io.quarkus.maven.ExtensionDescriptorMojoTest.shouldProcessRealisticExtensionCleanly(ExtensionDescriptorMojoTest.java:146)
    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at io.quarkus:quarkus-core:jar:2.12.0.Final -> org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:362)
    at io.quarkus.maven.ExtensionDescriptorMojo.resolveRuntimeDeps(ExtensionDescriptorMojo.java:887)
    ... 6 more
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at io.quarkus:quarkus-core:jar:2.12.0.Final -> org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final
    at org.eclipse.aether.internal.impl.collect.DependencyCollectorDelegate.collectDependencies(DependencyCollectorDelegate.java:260)
    at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:87)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:327)
    ... 7 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:289)
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:175)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.resolveCachedArtifactDescriptor(DfDependencyCollector.java:382)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.getArtifactDescriptorResult(DfDependencyCollector.java:368)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:218)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:156)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.process(DfDependencyCollector.java:138)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.doRecurse(DfDependencyCollector.java:343)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:277)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.processDependency(DfDependencyCollector.java:156)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.process(DfDependencyCollector.java:138)
    at org.eclipse.aether.internal.impl.collect.df.DfDependencyCollector.doCollectDependencies(DfDependencyCollector.java:108)
    at org.eclipse.aether.internal.impl.collect.DependencyCollectorDelegate.collectDependencies(DependencyCollectorDelegate.java:222)
    ... 9 more
Caused by: org.apache.maven.model.resolution.UnresolvableModelException: The following artifacts could not be resolved: org.jboss:jboss-parent:pom:34 (present, but unavailable): Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.jboss:jboss-parent:pom:34 has not been downloaded from it before.
    at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:160)
    at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:204)
    at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally(DefaultModelBuilder.java:1009)
    at org.apache.maven.model.building.DefaultModelBuilder.readParent(DefaultModelBuilder.java:801)
    at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:327)
    at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:243)
    at io.quarkus.bootstrap.resolver.maven.MavenModelBuilder.build(MavenModelBuilder.java:63)
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:284)
    ... 21 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.jboss:jboss-parent:pom:34 (present, but unavailable): Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.jboss:jboss-parent:pom:34 has not been downloaded from it before.
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:459)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:259)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:242)
    at org.apache.maven.repository.internal.DefaultModelResolver.resolveModel(DefaultModelResolver.java:158)
    ... 28 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.jboss:jboss-parent:pom:34 has not been downloaded from it before.
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:410)
    ... 31 more
Caused by: org.eclipse.aether.transfer.RepositoryOfflineException: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode
    at org.eclipse.aether.internal.impl.DefaultOfflineController.checkOffline(DefaultOfflineController.java:59)
    at org.eclipse.aether.internal.impl.Utils.checkOffline(Utils.java:106)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:404)
    ... 31 more
```

As a workaround, set the `io.quarkus.maven.ExtensionDescriptorMojoTest#RESOLVE_OFFLINE` field to `false` and re-run the tests in `independent-projects/extension-maven-plugin`. After the tests pass, the field can be reverted as the test will continue to pass from now on.

### Expected behavior

The test should pass.

### Actual behavior

The test fails with the aforementioned exception.

### How to Reproduce?

1. Start with a pristine system. It may be enough to drop `~/.m2/repository` but I haven't been brave enough to try it.
2. Build Quarkus, including this test.
3. Observe failure.

### Output of `uname -a` or `ver`

Darwin xxxx 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

openjdk version "17.0.9" 2023-10-17 OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9) OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode)

### Quarkus version or git rev

0d90b7c0b82262ed38a00ac8b34385f19c88dd9a

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546) Maven home: /Users/david/local/apache-maven Java version: 17.0.9, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "14.1.2", arch: "aarch64", family: "mac"

### Additional information

_No response_

--- a/extensions/panache/panacheql/.gitignore
+++ b/extensions/panache/panacheql/.gitignore
@@ -1,0 +1,3 @@
+gen/**
+**/internal/gen/**
+/src/main/antlr4/io/quarkus/panacheql/internal/HqlLexer.tokens

--- a/independent-projects/extension-maven-plugin/src/test/java/io/quarkus/maven/ExtensionDescriptorMojoTest.java
+++ b/independent-projects/extension-maven-plugin/src/test/java/io/quarkus/maven/ExtensionDescriptorMojoTest.java
@@ -142,7 +142,9 @@ class ExtensionDescriptorMojoTest extends AbstractMojoTestCase {
         mavenExecPom("fake-extension-runtime");
         mavenExecPom("fake-extension-deployment");
 
-        ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime");
+        // The test extension, by design, resolves dependencies of an older (and outside the main project) Quarkus version
+        // Usually it builds fine, even offline, but we cannot count on everything being already downloaded, especially
+        ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime", false);
         mojo.execute();
 
     }
@@ -156,7 +158,7 @@ class ExtensionDescriptorMojoTest extends AbstractMojoTestCase {
         mavenExecPom("fake-extension-deployment");
 
         Exception thrown = Assertions.assertThrows(MojoExecutionException.class, () -> {
-            ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime-with-missing-dependencies");
+            ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime-with-missing-dependencies", false);
             mojo.execute();
         });
         assertTrue("Message format does not match expectations: \n" + thrown.getMessage(),
@@ -178,7 +180,7 @@ class ExtensionDescriptorMojoTest extends AbstractMojoTestCase {
         mavenExecPom("fake-extension-deployment");
 
         Exception thrown = Assertions.assertThrows(MojoExecutionException.class, () -> {
-            ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime-with-missing-dependencies");
+            ExtensionDescriptorMojo mojo = makeMojo("fake-extension-runtime-with-missing-dependencies", false);
             mojo.execute();
         });
         String message = thrown.getMessage();
@@ -232,6 +234,10 @@ class ExtensionDescriptorMojoTest extends AbstractMojoTestCase {
     }
 
     private ExtensionDescriptorMojo makeMojo(String dirName) throws Exception {
+        return makeMojo(dirName, RESOLVE_OFFLINE);
+    }
+
+    private ExtensionDescriptorMojo makeMojo(String dirName, boolean resolveOffline) throws Exception {
         File basedir = getTestFile(TEST_RESOURCES + dirName);
         File pom = new File(basedir, "pom.xml");
 
@@ -245,7 +251,7 @@ class ExtensionDescriptorMojoTest extends AbstractMojoTestCase {
         final MavenArtifactResolver mvn = new MavenArtifactResolver(
                 new BootstrapMavenContext(BootstrapMavenContext.config()
                         .setCurrentProject(pom.getAbsolutePath())
-                        .setOffline(RESOLVE_OFFLINE)));
+                        .setOffline(resolveOffline)));
 
         ExtensionDescriptorMojo mojo = (ExtensionDescriptorMojo) lookupConfiguredMojo(session,
                 newMojoExecution("extension-descriptor"));

